### DISCUSSION TITLE
Full support of recipes with `MakieCore` (lift, legendelements, ATTRIBUTES...)

### DISCUSSION THREAD

**Discussion by BambOoxX** [JuliaPlots/Makie.jl#4221] [Ideas]
Unless I am mistaken, `MakieCore` is aimed at being the lightweight dependency package developpers should rely on when implementing specific recipes for their packages. Something alike `RecipesBase` for `Plots.jl`. 

While trying to switch from `Makie` to `MakieCore` as a dependency, I got an error showing that `@lift` is not defined in `MakieCore`, making use of observables surprinsingly impossible when relying on `MakieCore`.

A quick and dirty `idefined(MakieCore,lift)` returns a `false`, while the same on `Makie` is `true`. 

Is there something I'm missing here ? 

I'm on julia 1.8.0 using `Makie` v0.17.13 and `MakieCore` v0.4.0

**Comment by SimonDanisch** [comment]
Well, since it's lightweight, lots has been left out..Although I guess `@lift` could be added...

**Comment by BambOoxX** [comment]
@SimonDanisch BTW, I just checked the code in `MakieCore` and it seems that some comments refer to a solution involving `lift` e.g. here https://github.com/MakieOrg/Makie.jl/blob/c373d3f5b44b9c5eadf37293f03a89e88d26c4dd/MakieCore/src/attributes.jl#L110-L122
Does this  mean that `lift` alone, not the macro, should work ?

**Comment by BambOoxX** [comment]
@SimonDanisch While testing things around recipes, it seems that the `ATTRIBUTES` string lies in `Makie`, as well as `legendelements`. To me it feels like both could be good additions to `MakieCore`, but I may have missed equivalent stuff in there. 
What is your opinion on this ?

**Comment by BambOoxX** [comment]
@SimonDanisch I just checked the code a bit for `lift`. Do you think solving this would just require to move the `src/interaction` to `MakieCore` and reexport this in `Makie` ?

**Comment by BambOoxX** [comment]
Light Bump ! 👍

**Comment by SimonDanisch** [comment]
We can definitely move `const lift = map` to MakieCore!
I personally try to avoid `@lift` and now that we have `lift(f, scene/plot, observables...)` to bind the observable life time to a scene/plot object, which `@lift` doesn't support, I've been removing `@lift` from quite a few places.
So, I'm leaning against moving it to MakieCore ;)

**Comment by BambOoxX** [comment]
@SimonDanisch I think we can afford the `const lift=map` indeed ! What about `ATTRIBUTES` which are super handy for recipes (so `MakieCore` scope) and `legendelements` (since it feels weird to be able to write a recipe but not its legend, though its seems a more complicated modification) ?

**Comment by asinghvi17** [comment]
ATTRIBUTES pulls in DocStringExtensions.jl, which we may not want.

**Comment by BambOoxX** [comment]
Sure, if it is pulling another dependency, that is not good, although basic recipes like [`arrows`](https://github.com/MakieOrg/Makie.jl/blob/master/src/basic_recipes/arrows.jl) use `ATTRIBUTES`. So if this is not the proper way to make the help section for a recipe we are either missing a piece of code here or it should be in `MakieCore` shouldn't it ?

**Comment by BambOoxX** [comment]
> We can definitely move `const lift = map` to MakieCore! I personally try to avoid `@lift` and now that we have `lift(f, scene/plot, observables...)` to bind the observable life time to a scene/plot object, which `@lift` doesn't support, I've been removing `@lift` from quite a few places. So, I'm leaning against moving it to MakieCore ;)

@SimonDanisch Do you have an example of this in the documentation ? I could not find one

**Comment by BambOoxX** [comment]
Hi @SimonDanisch I read somewhere on the discourse I think, that you are close to updating things regarding recipes. Is there any news regarding this issue ?

**Comment by SimonDanisch** [comment]
Well, the new plotspec API is one part of the puzzle, another one will be the axis converts in [#442 ](https://github.com/MakieOrg/Makie.jl/pull/3226).
If that's all done I hope to clean up the missing bits and finally document everything nicely!
